cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -658,6 +662,18 @@ public static KafkaCluster fromCrd(Kafka kafkaAssembly, KafkaVersion.Lookup vers
         return result;
     }
 
+    private static void validateConfigProperty(String propertyName, KafkaClusterSpec kafkaClusterSpec) {
+        String orLess = kafkaClusterSpec.getReplicas() > 1 ? " or less" : "";
+        if (kafkaClusterSpec.getConfig() != null) {
+            if (kafkaClusterSpec.getConfig().get(propertyName) != null) {
+                if (Integer.parseInt(kafkaClusterSpec.getConfig().get(propertyName).toString()) > kafkaClusterSpec.getReplicas()) {
+                    log.warn("Kafka configuration '{}' should be set to {}{}.", propertyName, kafkaClusterSpec.getReplicas(), orLess);
+                    throw new InvalidResourceException("Kafka configuration '" + propertyName + "' should be set to " + kafkaClusterSpec.getReplicas() + orLess + ".");
```

Comment:
We should either warn or throw (which will only result in another log), not both so as to avoid duplicate log entries.